src/core/thread/mle.cpp

```diff
@@ -520,6 +520,7 @@ ThreadError Mle::SetRloc16(uint16_t aRloc16)
     }
 
     mMac.SetShortAddress(aRloc16);
+    mNetif.GetNetworkDataLocal().SendServerDataNotification();
```

Comment:
Good catch.  Do you think w still need a task if we simply call `SendServerDataNotification()` in `UpdateRoutes()`?